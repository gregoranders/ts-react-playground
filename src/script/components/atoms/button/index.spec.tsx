import React, { mount, create } from '@app/testUtils';

import * as TestSubject from '@atoms/button';

describe(`${TestSubject.Button.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  const OnClick = jest.fn();

  afterEach(() => {
    OnClick.mockReset();
  });

  it('snapshot', (): void => {
    const snapshot = create(<TestComponent onClick={OnClick} />);
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(<TestComponent onClick={OnClick} />);
    expect(testSubject.text()).toBe('Button');
  });

  it('click', () => {
    const testSubject = mount(<TestComponent onClick={OnClick} />);
    expect(testSubject.text()).toBe('Button');
    expect(OnClick.mock.calls.length).toEqual(0);
    testSubject.find('button').simulate('click');
    expect(OnClick.mock.calls.length).toEqual(1);
  });
});
