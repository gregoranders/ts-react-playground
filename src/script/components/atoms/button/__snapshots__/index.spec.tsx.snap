// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<button
  disabled={false}
  onClick={[Function]}
>
  Button
</button>
`;
